src/repositories/shopping-cart.repository.ts

```diff
@@ -42,6 +42,9 @@ export class ShoppingCartRepository extends DefaultKeyValueRepository<
    * @param userId User id
    * @param check A function that checks the current value and produces a new
    * value. It returns `null` to abort.
+   *
+   * @returns A promise of the updated ShoppingCart instance or `null` if the
```

Comment:
I find this sentence ambiguous. If the transaction fails, is the function returning `null` or a Promise that's resolved to `null`?

Also: A transaction can fail for other reasons besides a race condition, for example the server can become unreachable or unresponsive. We are not returning `null` but rejecting the promise in those cases!

Proposed wording:

_A promise that's resolved with the updated ShoppingCart instance or with `null` if the transaction failed because of a race condition._